utils/ccdatofhir/mapping_datatypes.bal

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// --------------------------------------------------------------------------------------------#
+// Source C-CDA to FHIR - Datatype Mappings
+// --------------------------------------------------------------------------------------------#
+
+import ballerina/log;
+import ballerinax/health.fhir.r4;
+
+xmlns "urn:hl7-org:v3" as v3;
+
+# Map C-CDA id to FHIR Identifier.
+#
+# + idElement - C-CDA id element
+# + return - Return FHIR Identifier
+public isolated function mapCcdaIdToFhirIdentifier(xml idElement) returns r4:Identifier? {
+    r4:Identifier identifier;
+    string?|error idVal = idElement.extension;
+    string?|error rootVal = idElement.root;
+    if rootVal is string {
```

Comment:
Check inverse and return early.

Reply:
re - With XML handling in Ballerina type error is thrown even the expected element is not available. But according to this use case, unavailability of certain elements(i.e, optional elements) is not a requirement to return with error but to proceed forward for other element mappings. That is the reason to check for type string, instead error.